docs/examples/rag_weaviate.ipynb

```diff
@@ -0,0 +1,1092 @@
+              "\u001b[1;32m│\u001b[0m everywhere, indicating it cannot distinguish between real and generated data. This framework allows for         \u001b[1;32m│\u001b[0m\n",
+              "\u001b[1;32m│\u001b[0m specific training algorithms and optimization techniques, such as backpropagation and dropout, to be            \u001b[1;32m│\u001b[0m\n",
+              "\u001b[1;32m│\u001b[0m effectively utilized.                                                                                           \u001b[1;32m│\u001b[0m\n",
+              "\u001b[1;32m╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# Create a prompt where context from the Weaviate collection will be injected\n",
+        "prompt = \"Explain how {text} works, using only the retrieved context.\"\n",
+        "query = \"a generative adversarial net\"\n",
+        "\n",
+        "response = collection.generate.near_text(\n",
+        "    query=query,\n",
+        "    limit=3,\n",
+        "    grouped_task=prompt,\n",
+        "    return_properties=[\"text\", \"title\"]\n",
+        ")\n",
+        "\n",
+        "# Prettify the output using Rich\n",
+        "console = Console()\n",
+        "\n",
+        "console.print(Panel(f\"{prompt}\".replace(\"{text}\", query), title=\"Prompt\", border_style=\"bold red\"))\n",
+        "console.print(Panel(response.generated, title=\"Generated Content\", border_style=\"bold green\"))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "7tGz49nfUegG"
+      },
+      "source": [
+        "We can see that our RAG pipeline performs relatively well for simple queries, especially given the small size of the dataset. Scaling this method for converting a larger sample of PDFs would require more compute (GPUs) and a more advanced deployment of Weaviate (like Docker, Kubernetes, or Weaviate Cloud). However, this notebook demonstrates that Docling is a robust and powerful open source tool for converting PDFs to structured data.\n",
+        "\n",
+        "To take this solution to next level, you consider:\n",
+        "* Experimenting with different [chunking techniques](https://weaviate.io/developers/academy/py/standalone/chunking/introduction)\n",
+        "* Using a RAG framework like [DSPy](https://weaviate.io/developers/integrations/llm-frameworks/dspy), [LlamaIndex](https://docs.llamaindex.ai/en/stable/api_reference/storage/vector_store/weaviate/), or [LangChain](https://python.langchain.com/docs/integrations/vectorstores/weaviate/)\n",
+        "* Implementing [advanced RAG](https://weaviate.io/blog/advanced-rag) techniques, like [Agentic RAG](https://weaviate.io/blog/what-is-agentic-rag)"
+      ]
```

Comment:
I assume these pointers are coming from a Weaviate-side notebook, where they would make more sense, but in this case, it would probably be best to provide a single link to Weaviate (e.g. docs) for wrapping up.

Reply:
Done.